public/app/plugins/datasource/cloud-monitoring/specs/datasource.test.ts

```diff
@@ -114,20 +114,12 @@ describe('CloudMonitoringDataSource', () => {
         expect(interpolated[2]).toBe('(filtervalue1|filtervalue2)');
       });
 
-      it('should not escape a regex', () => {
-        const templateSrv = initTemplateSrv('/[a-Z]*.html', true);
+      it('should escape regexes compatible with Google RE2 format', () => {
+        const templateSrv = initTemplateSrv(['/[a-Z]*.html', '/foo.html', '**/', '\\/'], true);
         const { ds } = getTestcontext({ templateSrv });
         const interpolated = ds.interpolateFilters(['resource.label.zone', '=~', '[[test]]'], {});
 
-        expect(interpolated[2]).toBe('/[a-Z]*.html');
-      });
-
-      it('should not escape an array of regexes but join them as a regex', () => {
-        const templateSrv = initTemplateSrv(['/[a-Z]*.html', '/foo.html'], true);
-        const { ds } = getTestcontext({ templateSrv });
-        const interpolated = ds.interpolateFilters(['resource.label.zone', '=~', '[[test]]'], {});
-
-        expect(interpolated[2]).toBe('(/[a-Z]*.html|/foo.html)');
+        expect(interpolated[2]).toBe('(/\\[a-Z\\]\\*\\.html|/foo\\.html|\\*\\*/|\\\\/)');
```

Comment:
This is an invalid query. If you try it you get: `unsupported escape sequence in a string literal at line 1, column 134, token '\\\"/\\\\[a-z\\\\]\\\\*\\\\.html\\\"'\",\n    \"status\": \"INVALID_ARGUMENT\"`.

If we were to follow this path, we would need to escape all special characters. Also, if you really wanted to use the character `.` as "any character", you wouldn't be able to do so.

Reply:
Closing in favour of another solution